src/utils/fetchBlocks.ts

```diff
@@ -0,0 +1,65 @@
+import { ethers } from 'ethers'
+import Block from '../models/block.model'
+import { logger } from './logger'
+import { getDateFromUnixTimestamp } from './util'
+
+const providerURL = process.env.PROVIDER_URL || 'https://rpc.gaiaxtestnet.oceanprotocol.com' // or use your local node 'http://localhost:8545'
+const provider = new ethers.providers.JsonRpcProvider(providerURL)
+
+async function getLatestBlockNumber() {
+  const latestBlock = await provider.getBlock('latest')
+  return latestBlock.number
+}
+
+async function getLatestBlockNumberFromDb() {
+  const blockArray = await Block.find({}).sort('-blockNumber').limit(1).exec()
+  return blockArray === [] ? null : blockArray[0].blockNumber
+}
+
+function getBlock(blockNumber) {
+  return provider.getBlock(blockNumber)
+}
+
+export async function fetchBlocks(bundleSize: number) {
+  logger.info('==== Start Block import ====')
+  try {
+    const latestBlockNumberInDb = await getLatestBlockNumberFromDb()
+    const latestBlockNumber = await getLatestBlockNumber()
+
+    logger.info(`Latest Block in DB: ${latestBlockNumberInDb} Lastest Block: ${latestBlockNumber}`)
+
+    let blockArray = []
+    let blockNumberStart = latestBlockNumberInDb ? latestBlockNumberInDb + 1 : 0
+    let blockNumberEnd = blockNumberStart + bundleSize - 1
+
+    for (let currentBlockNumber = blockNumberStart; currentBlockNumber < latestBlockNumber; currentBlockNumber++) {
+      if (currentBlockNumber > blockNumberEnd) {
+        const newBlockNumberEnd = blockNumberEnd + bundleSize
+        await Block.insertMany(blockArray)
+        logger.info(`blocks saved to DB (bundle size: ${bundleSize})`)
+        blockArray = []
+
+        blockNumberStart = blockNumberStart + bundleSize
+        blockNumberEnd = newBlockNumberEnd > latestBlockNumber ? latestBlockNumber : newBlockNumberEnd
+      }
+
+      const newBlock = await getBlock(currentBlockNumber)
+
+      const { number, timestamp, transactions }: { number: number; timestamp: number; transactions: string[] } = newBlock
+
+      logger.info(`Fetch Block: ${number.toLocaleString('en-US')} / ${latestBlockNumber.toLocaleString('en-US')}`)
+
+      blockArray.push({
+        blockNumber: number,
+        unixTimestamp: timestamp,
+        timestamp: getDateFromUnixTimestamp(timestamp),
+        transactionHashes: transactions
+      })
+    }
+    await Block.insertMany(blockArray)
```

Comment:
nitpick(blocking): Separation of concerns. Saving should not happen in the same function as fetching.